File: MyProjects/MAP/Assignment 7/Interpreter/Model/Statements/AssignStmt.cs
using System;
using System.Collections.Generic;
using Interpreter.Model.Expressions;
using System.Text;

namespace Interpreter.Model.Statements
{
    public class AssignStmt:IStmt
    {
        private string id;
        private IExpr expr;

        public AssignStmt(string key, IExpr expression)
        {
            id = key;
            expr = expression;
        }

        public ProgramState execute(ProgramState state)
        {
            int val = expr.evaluate(state.SymbolTable);
            state.SymbolTable.setValue(id, val);
            return state;
        }

        public override string ToString()
        {
            return id + "=" + expr;
        }
    }
}

File: MyProjects/MAP/Practical Exam/Procedures/src/Model/Statements/AssignStmt.java
package Model.Statements;

import Model.Expressions.*;
import Model.ProgramState;
import Model.Utilities.IDictionary;
import Model.Utilities.IStack;

public class AssignStmt implements IStmt {
    private String id;
    private IExpr expr;

    public AssignStmt(String key, IExpr expression){
        id = key;
        expr = expression;
    }

    @Override
    public ProgramState execute(ProgramState state) {
        IStack<IDictionary<String, Integer>> dic = state.getSyms();
        IDictionary<Integer, Integer> heap=state.getHeap();
        int val = expr.evaluate(dic,heap);
        dic.top().setValue(id, val);
        return null;
    }

    public String toString(){
        return id + "=" + expr;
    }
}
